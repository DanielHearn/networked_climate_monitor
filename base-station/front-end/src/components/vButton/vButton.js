export default {
  name: 'v-button',
  props: {
    type: {
      type: String,
      default: 'primary',
      required: true
    },
    text: {
      type: String,
      default: '',
      required: true
    },
    isIcon: {
      type: Boolean,
      default: false,
      required: false
    }
  },
  computed: {
    typeClass: function() {
      return `button--${this.type}`
    }
  }
}
