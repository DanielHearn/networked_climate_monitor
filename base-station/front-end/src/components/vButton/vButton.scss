@import "./../../scss/variables.scss";

.button {
  padding: calc(.5em - 2px) calc(.8em - 2px);
  cursor: pointer;
  font-family: $font-family;
  font-weight: $medium-weight;
  font-size: $s-size;
  -webkit-appearance: none;
  display: flex;
  justify-content: center;
  align-items: center;
  user-select: none;
  &--primary {
    color: $white;
    background: $accent;
    border: 2px solid $accent;
    transition: background $transition-fast;
    &:hover {
      background: $accent-dark;
    }
  }
  &--secondary {
    color: $white;
    background: $darkest;
    border: 2px solid $darkest;
    transition: background $transition-fast;
    &:hover {
      background: $darkester;
    }
  }
  &--tertiary {
    border: 2px solid $darkest;
    background: $white;
    color: $darkest;
    transition: background $transition-fast;
    &:hover {
      background: $grey-dark;
    }
  }
}