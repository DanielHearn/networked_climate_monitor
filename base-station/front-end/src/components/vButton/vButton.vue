<template>
  <button :class="['button', typeClass]">
    <template v-if="isIcon">
      <i class="material-icons">{{ text }}</i>
    </template>
    <template v-else>{{ text }}</template>
  </button>
</template>

<script src="./vButton.js"></script>

<style lang="scss" src="./vButton.scss"></style>
