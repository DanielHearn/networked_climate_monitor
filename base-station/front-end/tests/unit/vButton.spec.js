import { mount } from '@vue/test-utils'
import vButton from './../../src/components/vButton/vButton.vue'

describe('vButton.vue', () => {
  let wrapper

  afterEach(() => {
    wrapper.destroy()
  })

  it('Text Button', () => {
    let hierachyLevel = 'primary'
    const text = 'Click me'
    wrapper = mount(vButton, {
      propsData: {
        hierachyLevel: hierachyLevel,
        text: text
      }
    })

    expect(wrapper.text()).toBe(text)
    expect(wrapper.classes()).toContain(`button--${hierachyLevel}`)

    hierachyLevel = 'secondary'
    wrapper.setProps({ hierachyLevel: hierachyLevel })
    expect(wrapper.classes()).toContain(`button--${hierachyLevel}`)

    hierachyLevel = 'tertiary'
    wrapper.setProps({ hierachyLevel: hierachyLevel })
    expect(wrapper.classes()).toContain(`button--${hierachyLevel}`)
  })

  it('Icon Button', () => {
    let hierachyLevel = 'primary'
    const text = 'close'
    wrapper = mount(vButton, {
      propsData: {
        hierachyLevel: hierachyLevel,
        text: text,
        isIcon: true
      }
    })

    expect(wrapper.find('i').text()).toBe(text)
    expect(wrapper.classes()).toContain(`button--${hierachyLevel}`)

    hierachyLevel = 'secondary'
    wrapper.setProps({ hierachyLevel: hierachyLevel })
    expect(wrapper.classes()).toContain(`button--${hierachyLevel}`)

    hierachyLevel = 'tertiary'
    wrapper.setProps({ hierachyLevel: hierachyLevel })
    expect(wrapper.classes()).toContain(`button--${hierachyLevel}`)
  })

  it('has the expected html structure', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
