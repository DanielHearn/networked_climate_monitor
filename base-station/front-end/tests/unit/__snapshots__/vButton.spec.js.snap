// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vButton.vue has the expected html structure 1`] = `
<button
  class="button button--tertiary"
>
  <i
    class="material-icons"
  >
    close
  </i>
</button>
`;
